### ISSUE TITLE
3DS payments in iOS using >= 4.9.0 Hyperpay SDK

### ISSUE THREAD

**Issue by azharkhan1** [RMabroukS/react-native-hyperpay-sdk#6]
Does this package support 3ds payments for iOS on deploy version of SDK? Also I can see there's no code for it in native iOS in this package.

I am stuck in updating hyperpay sdk because new SDK has some new methods like onThreeDSChallengeRequiredWithCompletion which runs when we submit the transaction. As of now I am not able to get redirectURL in asynchronous payments and I can't process the payments due to that.

A help would be greatful, thanks

**Comment by RMabroukS** [comment]
@azharkhan1 resolved today on the latest version number 0.35.0

**Comment by azharkhan1** [comment]
@RMabroukS thanks for the quick response, can you please let me know what SDK did you use? I can see its 4.6.0. We need to use 4.9.0 or greater in our case. We can't seem to make 3ds payments work in iOS in any version greater or equal to 4.9.0. Can you please let me know if you were able to come up with any solution regarding this with the mentioned version?

**Comment by azharkhan1** [comment]
Never mind, We were able to solve this problem in our own codebase with the help of hyperpay support, we did not change anything in the code just added some 3ds related lines. Maybe they did some setting in their portal and it started working on production environment.

Thanks for the quite response, closing this issue as I did not use this package but since it had 4.9.0 SDK implementation I thought it would have the same issue! Cheers @RMabroukS

**Comment by azharkhan1** [comment]
@RMabroukS sure, we were not able to process with payments on iOS on version 4.5 as we used to receive alot of time out errors from hyperpay so they asked us to update it to 4.9, 4.9 did not work, then they gave us 5.1 SDK that did not work either. So then again they gave us 4.9 and it magically worked.

 Of course we did some changes in the code to support 3ds 2 payments but that was done at the time of updating SDK initially. Hope that helps.
